src/cryptography/hazmat/primitives/ciphers/algorithms.py

```diff
@@ -153,3 +153,16 @@ def nonce(self) -> bytes:
     @property
     def key_size(self) -> int:
         return len(self.key) * 8
+
+
+class SM4(CipherAlgorithm, BlockCipherAlgorithm):
+    name = "SM4"
+    block_size = 128
+    key_sizes = frozenset([128])
+
+    def __init__(self, key):
+        self.key = _verify_key_size(self, key)
+
+    @property
+    def key_size(self):
```

Comment:
```suggestion
    def key_size(self) -> int:
```

Reply:
Fixed, and the `__init__` parameters too.